jaeger/charts/linkerd-jaeger/templates/namespace-metadata.yaml

```diff
@@ -46,7 +53,19 @@ spec:
             ops="$ops{\"op\": \"add\",\"path\": \"/metadata/labels\",\"value\": {}},"
           fi
 
-          ops="$ops{\"op\": \"add\", \"path\": \"/metadata/labels/linkerd.io~1extension\", \"value\": \"jaeger\"}"
+          ops="$ops{\"op\": \"add\", \"path\": \"/metadata/labels/linkerd.io~1extension\", \"value\": \"jaeger\"},"
+
+          # grab the latest occurence of cniEnabled in linkerd-config, to
+          # discard value in the last-applied-configuration annotation
+          cniEnabled=$(curl -kfv -H "Authorization: Bearer $token" \
+            "https://kubernetes.default.svc/api/v1/namespaces/{{.Values.linkerdNamespace}}/configmaps/linkerd-config" | \
+            sed -r -n 's/.*cniEnabled: (\w+).*/\1/gp' | tail -1)
```

Comment:
is it worth pulling in something like yq here to parse this properly and make it a bit more robust?

Reply:
We'd have to use a different image than `curlImages/curl` to include anything besides `curl`. And given the privileges this has we can't install anything at runtime either. 
Ideally we could build a binary that did all this and include it in say the proxy image, but it'd require some extra maintenance.

Reply:
I've raised #9985 to track this effort separately.